### PR TITLE
Get Android NDK builds working on Windows

### PR DESCRIPTION
**Pull request by andrurogerz**
A set of minor patches put together with some help from @compnerd that fixes Android NDK cross-compilation on Windows.

Note: GitHub Actions to automatically perform this build will be in a subsequent pull request.

Test plan:

Passed existing GHA builds for Windows and Linux, MacOS is failing on the original branch: https://github.com/andrurogerz/ds2/actions/runs/9864533048

Locally built Android NDK with `CMAKE_ANDROID_ARCH_ABI=[x86, x86_64, armeabi-v7a, arm64-v8a]` on Windows using the following command:
`S:\ds2>set ANDROID_NDK_ROOT="C:/Program Files (x86)/Android/AndroidNDK/android-ndk-r23c"`
`S:\ds2>mkdir build`
`S:\ds2>cd build`
`S:\ds2\build>cmake -D CMAKE_SYSTEM_NAME=Android -DCMAKE_ANDROID_ARCH_ABI=arm64-v8a -DCMAKE_ANDROID_NDK=%ANDROID_NDK_ROOT% -G Ninja -S ..\ds2`
`S:\ds2\build>ninja`

The NDK used was installed with Android Studio.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -106,9 +106,9 @@ endif()
 
 get_property(GENERATOR_IS_MULTI_CONFIG GLOBAL PROPERTY GENERATOR_IS_MULTI_CONFIG)
 if(GENERATOR_IS_MULTI_CONFIG)
-  set(RegsGen2_RELATIVE_PATH Release/regsgen2${CMAKE_EXECUTABLE_SUFFIX})
+  set(RegsGen2_RELATIVE_PATH Release/regsgen2${CMAKE_HOST_EXECUTABLE_SUFFIX})
 else()
-  set(RegsGen2_RELATIVE_PATH regsgen2${CMAKE_EXECUTABLE_SUFFIX})
+  set(RegsGen2_RELATIVE_PATH regsgen2${CMAKE_HOST_EXECUTABLE_SUFFIX})
```
Comment (compnerd) [inline CMakeLists.txt:111]: This should be part of the change for the host exe suffix.

File: Tools/JSObjects/CMakeLists.txt [hunk 2]
```diff
@@ -58,6 +58,7 @@ if (CMAKE_CXX_COMPILER_ID MATCHES "GNU" OR
       COMPILE_FLAGS -Wno-error=unused-but-set-variable)
   target_compile_options(jsobjects PRIVATE -Wall -Wextra -Werror)
   target_compile_options(jsobjects PRIVATE -Wno-unused-parameter
+                                           -Wno-unknown-warning-option
```
Comment (compnerd) [inline Tools/JSObjects/CMakeLists.txt:61]: Can we limit this to Android or under a version check? If we can keep the clang versions homogenous, it makes it better for diagnosing portability issues to have all the warnings be similar.

File: CMakeLists.txt (hunk 1 above)
Comment (andrurogerz) [reply CMakeLists.txt:111]: Doh, bad split. I should have caught that. Will fix

**Comment by andrurogerz** [comment]
latest GHA build results after updates: https://github.com/andrurogerz/ds2/actions/runs/9864533048

File: Tools/JSObjects/CMakeLists.txt [hunk 3]
```diff
@@ -54,8 +54,15 @@ target_include_directories(jsobjects
...
-  set_source_files_properties(${JSObjects_BINARY_DIR}/parser.c PROPERTIES
-      COMPILE_FLAGS -Wno-error=unused-but-set-variable)
+
+  if ((CMAKE_C_COMPILER_ID MATCHES "GNU" AND CMAKE_C_COMPILER_VERSION VERSION_GREATER_EQUAL 4.6) OR
+      (CMAKE_C_COMPILER_ID MATCHES "Clang" AND CMAKE_C_COMPILER_VERSION VERSION_GREATER_EQUAL 13.0))
+    # this warning is enabled by -Wall on newer compilers and must be disabled
+    # for the generated parser code
+    set_source_files_properties(${JSObjects_BINARY_DIR}/parser.c PROPERTIES
+        COMPILE_FLAGS -Wno-error=unused-but-set-variable)
+  endif()
```
Comment (compnerd) [inline Tools/JSObjects/CMakeLists.txt:64]: I think that we can condense this a bit ...

```suggestion
if ((CMAKE_C_COMPILER_ID MATCHES "GNU" AND CMAKE_C_COMPILER_VERSION VERSION_GREATER_EQUAL 4.6) OR
    ((CMAKE_C_COMPILER_ID_MATCHES "Clang" AND NOT CMAKE_C_SIMULATE_ID MATCHES "MSVC") AND CMAKE_C_COMPILER_VERSION VERSION_GREATER_EQUAL 13.0)
  set_source_files_properties(${JSObjects_BINARY_DIR}/parser.c PROPERTIES
    COMPILE_FLAGS -Wno-error=unused-but-set-variable)
```

**Review by compnerd — APPROVED** [review]

File: Tools/JSObjects/CMakeLists.txt (hunk 3 above)
Comment (compnerd) [reply Tools/JSObjects/CMakeLists.txt:64]: I guess we can do this as a follow up... (there was another clean possibility up that I noticed).

**Pull request merged**